### DISCUSSION TITLE
Can T61STRING, TELETEXSTRING, GRAPHICSTRINGs be replaced with UTF8STRINGs?

### DISCUSSION THREAD

**Discussion by naveedpash** [librasn/rasn#60] [Q&A] [answered]
Hey, during my project, I'm running into a bunch of arcane ASN.1 string types and was wondering if it's ok to just substitute them all with UTF8String?

I keep reading at random places on the internet that they're all deprecated in favor of UTF8String but need someone else to confirm this...

**Comment by vincentdephily** [comment, marked as answer]
If you're creating a new format, by all means use UTF8String (or a restricted ascii set if you're not dealing with human text : UPER can squeeze each char into less than 8 bits).

If you're implementing an existing format, you can't change its grammar. I'm not sure how well `rasn` supports those; decoding to utf8 is probably infallible, but not encoding.

**Reply by XAMPPRocky** [reply]
Yes, @vincentdephily is correct, new specs should use `Utf8String`, you still need to correctly handle old spec's strings, as their could be machines still using those encodings sending you messages. `rasn` currently doesn't handle all of the restrictions of the various strings for you, because that would require making a bunch of custom string types is a bunch of work, and no one's been asking for them. Eventually I'd like to replace all of the implicit wrappers with custom string types, but I haven't gotten around to it.
